### DISCUSSION TITLE
Create Role Collections during subaccount setup

### DISCUSSION THREAD

**Discussion by AnatolManikalo** [SAP-archive/btp-setup-automator#413] [Q&A] [answered]
Hello,
Could you please advise how I can declare Role Collection creation with some roles inside?

I tried adding the following object to **"assignrolecollections"** section:

`{
        "name": "L2 Support",
        "type": "account",
        "level": "sub account",
		"assignedRoles": [
			"Cloud Connector Administrator",
			"Destination Administrator",
			"Subaccount Admin",
			"Subaccount Service Administrator",
			"User and Role Administrator"
		],
        "assignedUserGroupsFromParameterFile": [
          
          "groupname"
        ]
	}`

But when running the script, I'm getting the following error:

[2023-01-23 14:48:19] ERROR      : 
[2023-01-23 14:48:19] ERROR      : {
  "error": "Role collection 'L2 Support' does not exist."
}

FAILED

I checked https://raw.githubusercontent.com/SAP-samples/btp-setup-automator/main/libs/btpsa-usecase.json and found that "services" section has

"assignrolecollections": {
"type": "array",
"description": "**list of role collections to be created within the btp account**",
"title": "list of role collections to be created within the btp account",
"default": []
}

So I tried adding this line to "services", but it didn't work out:

"services": [
    {
      "name": "cloudfoundry",
      "plan": "standard",
      "category": "ENVIRONMENT",
	  "assignrolecollections": [
		  "L2 Support"
	  ]
    }
  ]

Could you please advise?

**Comment by lechnerc77** [comment, marked as answer]
You find an example of the assignment of a role collection here: <https://github.com/SAP-samples/btp-setup-automator/blob/main/usecases/released/default.json>.

In your case I assume that this is a custom role collection and I guess you need to add the type "custom" in your configuration instead of account
 
The code and different option executed are contained in the file <https://github.com/SAP-samples/btp-setup-automator/blob/main/libs/python/helperRolesAndUsers.py>, which gives you an insight into the different steps that are executed.
For custom role collections you find the flow in the code [here](https://github.com/SAP-samples/btp-setup-automator/blob/main/libs/python/helperRolesAndUsers.py#LL222C52-L222C52)

**Reply by AnatolManikalo** [reply]
Thanks!
adding "custom" worked :)
